### PR TITLE
Synapse dataplane

### PR DESCRIPTION
**Pull request by lmazuel**
With autorest v5

### CODE REVIEW

File: sdk/synapse/azure-synapse/azure/synapse/_configuration.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    :param synapse_dns_suffix: Gets the DNS suffix used as the base for all Synapse service requests.
+    :type synapse_dns_suffix: str
+    :param livy_api_version: Valid api-version for the request.
+    :type livy_api_version: str
+    """
+
+    def __init__(
+        self,
+        credential,  # type: "TokenCredential"
+        synapse_dns_suffix,  # type: str
```
Comment (aim-for-better) [inline sdk/synapse/azure-synapse/azure/synapse/_configuration.py]: Can we set default value "dev.azuresynapse.net" for synapse_dns_suffix? Just like livy_api_version

File: sdk/synapse/azure-synapse/azure/synapse/_synapse_client.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    :param synapse_dns_suffix: Gets the DNS suffix used as the base for all Synapse service requests.
+    :type synapse_dns_suffix: str
+    :param livy_api_version: Valid api-version for the request.
+    :type livy_api_version: str
+    """
+
+    def __init__(
+        self,
+        credential,  # type: "TokenCredential"
+        synapse_dns_suffix,  # type: str
```
Comment (aim-for-better) [inline sdk/synapse/azure-synapse/azure/synapse/_synapse_client.py]: Can we set default value "dev.azuresynapse.net" for synapse_dns_suffix? Just like livy_api_version

File: sdk/synapse/azure-synapse/azure/synapse/aio/_configuration_async.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    :param synapse_dns_suffix: Gets the DNS suffix used as the base for all Synapse service requests.
+    :type synapse_dns_suffix: str
+    :param livy_api_version: Valid api-version for the request.
+    :type livy_api_version: str
+    """
+
+    def __init__(
+        self,
+        credential: "TokenCredential",
+        synapse_dns_suffix: str,
```
Comment (aim-for-better) [inline sdk/synapse/azure-synapse/azure/synapse/aio/_configuration_async.py]: Can we set default value "dev.azuresynapse.net" for synapse_dns_suffix? Just like livy_api_version

File: sdk/synapse/azure-synapse/azure/synapse/aio/_synapse_client_async.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    :param synapse_dns_suffix: Gets the DNS suffix used as the base for all Synapse service requests.
+    :type synapse_dns_suffix: str
+    :param livy_api_version: Valid api-version for the request.
+    :type livy_api_version: str
+    """
+
+    def __init__(
+        self,
+        credential: "TokenCredential",
+        synapse_dns_suffix: str,
```
Comment (aim-for-better) [inline sdk/synapse/azure-synapse/azure/synapse/aio/_synapse_client_async.py]: Can we set default value "dev.azuresynapse.net" for synapse_dns_suffix? Just like livy_api_version

File: sdk/synapse/azure-synapse/azure/synapse/_configuration.py (hunk 1 above)
Comment (lmazuel) [reply sdk/synapse/azure-synapse/azure/synapse/_configuration.py]: fixed

File: sdk/synapse/azure-synapse/azure/synapse/aio/_configuration_async.py (hunk 3 above)
Comment (lmazuel) [reply sdk/synapse/azure-synapse/azure/synapse/aio/_configuration_async.py]: fixed

File: sdk/synapse/azure-synapse/azure/synapse/aio/_synapse_client_async.py (hunk 4 above)
Comment (lmazuel) [reply sdk/synapse/azure-synapse/azure/synapse/aio/_synapse_client_async.py]: fixed

File: sdk/synapse/azure-synapse/azure/synapse/_synapse_client.py (hunk 2 above)
Comment (lmazuel) [reply sdk/synapse/azure-synapse/azure/synapse/_synapse_client.py]: fixed

**Pull request merged**